src/App.tsx

```diff
@@ -1,14 +1,70 @@
-/* eslint-disable max-len */
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+const getVisibleTodos = (
+  todos: Todo[],
+  sortType: string,
+  query: string,
+): Todo[] => {
+  let categoryFiltered = todos;
+
+  if (sortType === 'active') {
+    categoryFiltered = todos.filter(todo => !todo.completed);
+  }
+
+  if (sortType === 'completed') {
+    categoryFiltered = todos.filter(todo => todo.completed);
+  }
+
+  return categoryFiltered.filter(todo => (
+    todo.title.toLowerCase().includes(query.toLowerCase())
+  ));
+};
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeTodo, setActiveTodo] = useState<Todo | null>(null);
```

Comment:
wow, again missed something, why do we save an entire activeTodo, when we can just save it's `id`?

Reply:
why, I send whole todo as props for modal popup